tpcc/delivery.go

```diff
@@ -48,87 +49,64 @@ func (del delivery) run(db *sql.DB, wID int) (interface{}, error) {
 	oCarrierID := rand.Intn(10) + 1
 	olDeliveryD := time.Now()
 
-	if err := crdb.ExecuteTx(
-		context.Background(),
-		db,
-		txOpts,
-		func(tx *sql.Tx) error {
-			getNewOrder, err := tx.Prepare(`
-			SELECT no_o_id
-			FROM new_order
-			WHERE no_w_id = $1 AND no_d_id = $2
-			ORDER BY no_o_id ASC
-			LIMIT 1`)
-			if err != nil {
-				return err
-			}
-			delNewOrder, err := tx.Prepare(`
-			DELETE FROM new_order
-			WHERE no_w_id = $1 AND no_d_id = $2 AND no_o_id = $3`)
-			if err != nil {
-				return err
-			}
-			updateOrder, err := tx.Prepare(`
-			UPDATE "order"
-			SET o_carrier_id = $1
-			WHERE o_w_id = $2 AND o_d_id = $3 AND o_id = $4
-			RETURNING o_c_id`)
-			if err != nil {
-				return err
-			}
-			updateOrderLine, err := tx.Prepare(`
-			UPDATE order_line
-			SET ol_delivery_d = $1
-			WHERE ol_w_id = $2 AND ol_d_id = $3 AND ol_o_id = $4`)
-			if err != nil {
-				return err
-			}
-			sumOrderLine, err := tx.Prepare(`
-			SELECT SUM(ol_amount) FROM order_line
-			WHERE ol_w_id = $1 AND ol_d_id = $2 AND ol_o_id = $3`)
-			if err != nil {
-				return err
-			}
-			updateCustomer, err := tx.Prepare(`
-			UPDATE customer
-			SET (c_balance, c_delivery_cnt) =
-				(c_Balance + $1, c_delivery_cnt + 1)
-			WHERE c_w_id = $2 AND c_d_id = $3 AND c_id = $4`)
-			if err != nil {
-				return err
-			}
-
-			// 2.7.4.2. For each district:
-			for dID := 1; dID <= 10; dID++ {
+	// 2.7.4.2. For each district:
+	for dID := 1; dID <= 10; dID++ {
+		if err := crdb.ExecuteTx(
+			context.Background(),
+			db,
+			txOpts,
+			func(tx *sql.Tx) error {
 				var oID int
-				if err := getNewOrder.QueryRow(wID, dID).Scan(&oID); err != nil {
+				if err := tx.QueryRow(fmt.Sprintf(`
+						SELECT no_o_id
+						FROM new_order
+						WHERE no_w_id = %d AND no_d_id = %d
+						ORDER BY no_o_id ASC
+						LIMIT 1`,
+					wID, dID)).Scan(&oID); err != nil {
 					// If no matching order is found, the delivery of this order is skipped.
 					if err != sql.ErrNoRows {
 						return err
 					}
-					continue
+					return nil
 				}
-				if _, err := delNewOrder.Exec(wID, dID, oID); err != nil {
+				if _, err := tx.Exec(fmt.Sprintf(`
+						DELETE FROM new_order
+						WHERE no_w_id = %d AND no_d_id = %d AND no_o_id = %d`,
+					wID, dID, oID)); err != nil {
```

Comment:
I think we can combine the `DELETE FROM new_order` with the previous `SELECT` using `DELETE FROM new_order WHERE no_w_id = %d AND no_d_id = %d ORDER BY no_o_id ASC LIMIT 1 RETURNING  no_o_id`. Would definitely need to test this.